test/e2e/e2e_test.go

```diff
@@ -128,10 +135,61 @@ func TestMain(m *testing.M) {
 		testfiles.AddFileSource(testfiles.RootFileSource{Root: framework.TestContext.RepoRoot})
 	}
 
+	if framework.TestContext.PrepullImages {
+		klog.Infof("Pre-pulling images so that they are cached for the tests.")
+		prepullImages()
+	}
+
 	rand.Seed(time.Now().UnixNano())
 	os.Exit(m.Run())
 }
 
 func TestE2E(t *testing.T) {
 	RunE2ETests(t)
 }
+
+func prepullImages() {
+	// These are typically set by ginkg.It, but we're running these beforehand.
+	gomega.RegisterFailHandler(ginkgo.Fail)
+	defer ginkgo.GinkgoRecover()
+
+	f := framework.NewDefaultFramework("image-prepull")
+	f.BeforeEach()
+	defer f.AfterEach()
+	ns := f.Namespace.Name
+	c := f.ClientSet
+
+	prePulledImages := []int{
+		image.Agnhost,
+		image.BusyBox,
+		image.Nginx,
+		image.Httpd,
+	}
```

Comment:
This seems like something someone will want to override or change eventually.  Would it make more sense as a top level var, the way node e2e's do this? I'm pretty ambivalent here, I just doubt this will be the final setting for this